components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -669,6 +670,21 @@ void AIChatTabHelper::MakeAPIRequestWithConversationHistoryUpdate(
                             std::move(data_received_callback));
 }
 
+void AIChatTabHelper::RetryAPIRequest() {
+  SetAPIError(mojom::APIError::None);
+  DCHECK(!chat_history_.empty());
+
+  std::vector<ConversationTurn>::reverse_iterator rit = chat_history_.rbegin();
+
+  for (; rit != chat_history_.rend(); ++rit) {
+    if ((*rit).character_type == CharacterType::HUMAN) {
+      ConversationTurn turn_copy = (*rit);
+      MakeAPIRequestWithConversationHistoryUpdate(turn_copy);
+      break;
+    }
```

Comment:
@petemill This makes another request with the last human entry however this duplicates it too. Essentially, we'd want to retry without duping, right?

Reply:
`turn_copy` is a little misleading because it's not a copy, right? I suggest removing the turn from history, and removing any partial response after if it exists, and then submitting the turn again.

Reply:
updated